__tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx

```diff
@@ -0,0 +1,31 @@
+import ProgressForm from '@/components/ProgressForm/ProgressForm';
+import { renderWithProviders } from '@/test-utils/renderWithProvider';
+import { fireEvent, screen } from '@testing-library/react';
+
+describe('Progress form', function () {
+    it('Should Render 3 input fields with appropriate data', function (){
+        renderWithProviders(<ProgressForm/>);
+        const textBoxes = screen.getAllByRole('textbox');
+        expect(textBoxes).toHaveLength(3);
+        const button = screen.getByRole('button') as HTMLButtonElement;
+        expect(button).toBeInTheDocument();
+        expect(button.type).toBe('submit');
```

Comment:
Suggestion: you can also write like this
`const button = screen.getByRole('button', {type: /submit/})`

Reply:
As per the role API reference, there is no property named type

Official docs:
https://testing-library.com/docs/queries/byrole/#api

Reply:
yes I mixed it up name , okay resolved